### PR TITLE
fix(train): dynamically detect GPU capability and look up peak FLOPS for MFU calculation

### LINKED ISSUE #5: MFU calculation is hardcoded to H100 peak FLOPS
## Problem

After PR #2 added support for non-Hopper GPUs, the MFU (Model FLOPS Utilization) metric is now misleading on any GPU that isn't an H100.

The peak FLOPS value is hardcoded at line 491:
\\\python
H100_BF16_PEAK_FLOPS = 989.5e12
\\\

On an A100 (312 TFLOPS BF16), the reported MFU would be ~3.2x too high. On an RTX 4090 (165 TFLOPS), it would be ~6x too high. This makes the MFU metric useless for comparing runs across different hardware.

## Proposed Fix

Auto-detect the GPU via \	orch.cuda.get_device_capability()\ and look up the correct theoretical peak FLOPS from a table of known GPU architectures. Fall back to the H100 value for unknown GPUs.

This is a minimal change (~10 lines) that makes the logging output accurate on all supported hardware.

I'll submit a PR for this.

### PR DESCRIPTION
**Pull request by aniruddhaadak80**
Resolves #5. The Model FLOPS Utilization (MFU) calculation previously assumed hardcoded H100 peak BF16 Tensor FLOPS (989.5 TFLOPS). This change detects the active GPU device name using torch.cuda.get_device_name(0), and sets the theoretical peak FLOPS using a lookup table for common GPUs (H100, A100, L40S, L4, RTX 4090, RTX 3090, A10G, etc.), falling back to the H100 peak value for unknown devices. This ensures MFU is accurate and comparable across different hardware environments.

### CODE REVIEW

**Review by svlandeg — COMMENTED** [review]
`nanochat` has this spelled out in more detail, cf https://github.com/karpathy/nanochat/blob/dc54a1a3077cab11d68fac4c5d1cd5c51f5d8c7a/nanochat/common.py#L227, but I think the idea for `autoresearch` was to provide a more minimal training loop to allow for quick experiments. Users can fork and adjust this as they please.
